onnx/shape_inference/implementation.cc

```diff
@@ -130,12 +175,43 @@ void mergeShapesAndTypes(const TypeProto_Tensor& inferredType, TypeProto_Tensor*
   }
 }
 
+void mergeShapesAndTypes(const TypeProto_SparseTensor& inferredType, TypeProto_SparseTensor* existingType) {
+  if (existingType->elem_type() == TensorProto::UNDEFINED) {
+    existingType->set_elem_type(inferredType.elem_type());
+  }
+
+  if (!inferredType.has_shape()) {
+    return;
+  }
+
+  if (!existingType->has_shape()) {
+    // Ensure the shape is initialized. Note that this must be done
+    // even for (zero-dimensional) scalars.
+    existingType->mutable_shape();
+
+    for (int j = 0; j < inferredType.shape().dim_size(); ++j) {
+      existingType->mutable_shape()->add_dim();
```

Comment:
if existingType does not have shape info then you can simply copy the shape info from inferredType there and return... no need to enter the for loop on line 197 right?

Reply:
I agree, but the existing code favors this form.